tests/foreman/cli/test_subscription.py

```diff
@@ -189,3 +241,475 @@ def test_negative_manifest_refresh(self):
+        new_subscription = None
+        for sub in org_subscriptions:
+            if sub['Subscription Name'] == SATELLITE_SUBSCRIPTION_NAME:
+                new_subscription = sub
+                break
+        self.assertIsNotNone(new_subscription)
+        # retrieve the default subscription id
+        org_subs = Subscription.list({u'organization-id': self.org['id']})
+        default_subscription_id = None
+        for sub in org_subs:
+            if sub['name'] == DEFAULT_SUBSCRIPTION_NAME:
+                default_subscription_id = sub['id']
+                break
+        self.assertIsNotNone(
+            default_subscription_id, msg='Default subscription not found')
+        # create 2 Virtual machines
+        with VirtualMachine() as client1, VirtualMachine() as client2:
+            hosts = []
+            for client in [client1, client2]:
+                client.install_katello_ca()
+                client.register_contenthost(
+                    self.org['label'], activation_key=activation_key['name'])
+                self.assertTrue(client.subscribed)
+                host = Host.info({'name': client.hostname})
+                hosts.append(host)
+                Host.subscription_attach({
+                    'host-id': host['id'],
+                    'subscription-id': default_subscription_id,
+                })
+                host_subscriptions = ActivationKey.subscriptions({
+                    'organization-id': self.org['id'],
+                    'id': activation_key['id'],
+                    'host-id': host['id'],
+                }, output_format='json')
+                self.assertEqual(len(host_subscriptions), 1)
+                self.assertEqual(
+                    host_subscriptions[0]['name'], DEFAULT_SUBSCRIPTION_NAME)
+            # export the content host data to csv file
+            chs_export_file_name = 'chs_export_{0}.csv'.format(
+                self.org['label'])
+            chs_export_remote_file_path = (
+                '/tmp/{0}'.format(chs_export_file_name)
+            )
+            CSV_.content_hosts({
+                'export': True,
+                'file': chs_export_remote_file_path,
+                'organization': self.org['name'],
+            })
+            # download the csv file
+            chs_export_local_file_path = os.path.join(
+                tempfile.gettempdir(), chs_export_file_name)
+            download_file(
+                chs_export_remote_file_path, chs_export_local_file_path)
+            # modify the content hosts subscription
+            field_names, csv_data = self._read_csv_file(
+                chs_export_local_file_path)
+            # each client is represented by one row of data
+            self.assertEqual(len(csv_data), 2)
+            for row_data in csv_data:
+                # The subscription is saved in the following format:
+                # """<quantity>|<sku>|<name>|<contract>|<account>"""
```

Comment:
bonus karma for such comments, they're very helpful for understanding the code 👍